lib/internal/modules/esm/worker.js

```diff
@@ -0,0 +1,35 @@
+const { ESMLoader } = require('internal/modules/esm/loader');
+const { workerData } = require('internal/worker_threads');
+const { deserialize, serialize } = require('v8');
+
+const { commsChannel } = workerData;
+// lock = 0 → main sleeps
+// lock = 1 → worker sleeps
+const lock = new Int32Array(commsChannel, 0, 4); // Required by Atomics
+const requestResponseData = new Uint8Array(commsChannel, 4, 2044); // for TextEncoder/Decoder
+
+const publicESMLoader = new ESMLoader();
+
+const types = {
+  'addCustomLoaders': publicESMLoader.addCustomLoaders,
+  'cjsCache.delete': publicESMLoader.cjsCache.delete,
+  'cjsCache.get': publicESMLoader.cjsCache.get,
+  'cjsCache.has': publicESMLoader.cjsCache.has,
+  'cjsCache.set': publicESMLoader.cjsCache.set,
+  'import': publicESMLoader.import,
+  'resolve': publicESMLoader.resolve,
+};
+
+Atomics.store(lock, 0, 1); // Send 'ready' signal to main
+Atomics.notify(lock, 0); // Notify main of signal
+
+while (true) { // event loop
```

Comment:
I think it's not a good idea to call that "event loop" when event loop already means something else in JS-land.
```suggestion
while (true) {
```